controller/service.go

```diff
@@ -214,3 +257,37 @@ func (c *controller) allocateIPs(key string, svc *v1.Service) ([]net.IP, error)
 	// Okay, in that case just bruteforce across all pools.
 	return c.ips.Allocate(key, serviceIPFamily, k8salloc.Ports(svc), k8salloc.SharingKey(svc), k8salloc.BackendKey(svc))
 }
+
+func (c *controller) getDesiredLbIPs(svc *v1.Service) ([]net.IP, error) {
+	lbIPsStr := svc.Annotations[annotationLoadBalancerIPs]
+	if lbIPsStr == "" {
+		return nil, nil
+	}
+	var lbIPs []net.IP
+	desiredLbIPs := strings.Split(lbIPsStr, ",")
+	for _, desiredLbIP := range desiredLbIPs {
+		ip := net.ParseIP(desiredLbIP)
+		if ip == nil {
+			return nil, fmt.Errorf("invalid %s: %q", annotationLoadBalancerIPs, lbIPsStr)
+		}
+		lbIPs = append(lbIPs, ip)
+	}
+	return lbIPs, nil
+}
+
+func (c *controller) isLbIPsEqual(ipsA []net.IP, ipsB []net.IP) (bool, error) {
```

Comment:
Same, no need to be a method.

Reply:
Also, rename as `ipsAreEqual` because that's what the function do.

Reply:
removed this method and using `sort` and `reflect.DeepEqual `now